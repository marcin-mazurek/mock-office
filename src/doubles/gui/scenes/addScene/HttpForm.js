import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { init as initAddScene } from './actions';
import { getSelectedServerDetails } from '../../entities/servers/selectors';

export class HttpForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleRequirementsChange = this.handleRequirementsChange.bind(this);
    this.handleDelayChange = this.handleDelayChange.bind(this);
    this.state = {};
  }

  handleRequirementsChange(event) {
    this.setState({
      requirements: event.target.value
    });
  }

  handleDelayChange(event) {
    this.setState({
      delay: parseInt(event.target.value, 10)
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const delay = parseInt(this.state.delay, 10);
    let requirements;

    try {
      if (this.state.requirements) {
        requirements = JSON.parse(this.state.requirements);
      }
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    this.props.initAddScene(this.props.queueId, {
      delay,
      requirements
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <label htmlFor="delay">Delay:</label>
          <input type="number" name="delay" onChange={this.handleDelayChange} />
        </div>
        <div>
          <label htmlFor="requirements">Requirements:</label>
          <div>
            <textarea
              name="requirements"
              cols="30"
              rows="10"
              onChange={this.handleRequirementsChange}
            />
          </div>
        </div>
        <div>
          <label htmlFor="payload">Payload:</label>
          <div>
            <textarea name="payload" cols="30" rows="10" onChange={this.handlePayloadChange} />
          </div>
        </div>
        <div>
          <button type="submit">
            Submit
          </button>
        </div>
      </form>
    );
  }
}

HttpForm.propTypes = {
  initAddScene: PropTypes.func.isRequired,
  queueId: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  queueId: getSelectedServerDetails(state).queue
});

const mapDispatchToProps = {
  initAddScene
};

export default connect(mapStateToProps, mapDispatchToProps)(HttpForm);
