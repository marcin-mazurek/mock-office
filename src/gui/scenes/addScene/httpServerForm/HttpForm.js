import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm, formValueSelector, FormSection } from 'redux-form/immutable';
import { connect } from 'react-redux';
import ScenePartFormSection from '../ScenePartFormSection';
import { submitHttpScene as createSubmitHttpSceneAction } from './actions';

export const HttpForm = props => (
  <form className="form" onSubmit={props.handleSubmit}>
    <div className="form__header">
      Add scene
    </div>
    <div className="form-row">
      <div className="form__field">
        <label className="form-field__label" htmlFor="title">Title:</label>
        <Field
          name="title"
          component="input"
          type="text"
          placeholder="Scene title"
          className="form-field__input"
        />
      </div>
    </div>
    <div className="form-row">
      <div className="form__field">
        <label className="form-field__label" htmlFor="requirements">Requirements:</label>
        <Field
          name="requirements"
          component="textarea"
          cols="30"
          rows="10"
        />
      </div>
    </div>
    <div>Part:</div>
    <FormSection name="scenePart">
      <ScenePartFormSection />
    </FormSection>
    <div className="form-row">
      <button className="button form__button" type="submit">
        Submit
      </button>
    </div>
  </form>
);

HttpForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

HttpForm.defaultPropTypes = {
  needRequirements: false
};

const selector = formValueSelector('httpServerScene');

const mapStateToProps = state => ({
  needRequirements: selector(state, 'type')
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  onSubmit: values => dispatch(createSubmitHttpSceneAction(ownProps.scenarioId, values))
});

export default connect(mapStateToProps, mapDispatchToProps)(
  reduxForm(
    {
      form: 'httpServerScene',
      initialValues: {
        scenePart: {
          type: 'immediate'
        }
      }
    }
  )(HttpForm)
);
