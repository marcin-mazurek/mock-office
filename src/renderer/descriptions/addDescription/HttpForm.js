import React from 'react';
import { connect } from 'react-redux';
import { init as initAddDescription } from './actions';
import { getSelectedServerDetails } from '../../servers/selectors';

export class HttpForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleBlockingChange = this.handleBlockingChange.bind(this);
    this.handleRequirementsChange = this.handleRequirementsChange.bind(this);
    this.handlePayloadChange = this.handlePayloadChange.bind(this);
    this.handleDelayChange = this.handleDelayChange.bind(this);
    this.state = {};
  }

  handleBlockingChange(event) {
    this.setState({
      blocking: event.target.checked
    });
  }

  handleRequirementsChange(event) {
    this.setState({
      requirements: event.target.value
    });
  }

  handlePayloadChange(event) {
    this.setState({
      descriptionPayload: event.target.value
    });
  }

  handleDelayChange(event) {
    this.setState({
      delay: parseInt(event.target.value, 10)
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const delay = parseInt(this.state.delay, 10);
    let requirements;

    try {
      if (this.state.requirements) {
        requirements = JSON.parse(this.state.requirements);
      }
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    let descriptionPayload;

    try {
      descriptionPayload = JSON.parse(this.state.descriptionPayload);
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    this.props.initAddDescription(this.props.queueId, {
      blocking: this.state.blocking,
      delay,
      requirements,
      descriptionPayload
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <label htmlFor="blocking">Blocking:</label>
          <input
            type="checkbox"
            name="blocking"
            onChange={this.handleBlockingChange}
            value={this.state.blocking}
          />
        </div>
        <div>
          <label htmlFor="delay">Delay:</label>
          <input type="number" name="delay" onChange={this.handleDelayChange} />
        </div>
        <div>
          <label htmlFor="requirements">Requirements:</label>
          <div>
            <textarea
              name="requirements"
              cols="30"
              rows="10"
              onChange={this.handleRequirementsChange}
            />
          </div>
        </div>
        <div>
          <label htmlFor="payload">Payload:</label>
          <div>
            <textarea name="payload" cols="30" rows="10" onChange={this.handlePayloadChange} />
          </div>
        </div>
        <div>
          <button type="submit">
            Submit
          </button>
        </div>
      </form>
    );
  }
}

HttpForm.propTypes = {
  initAddDescription: React.PropTypes.func.isRequired,
  queueId: React.PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  queueId: getSelectedServerDetails(state).queue
});

const mapDispatchToProps = {
  initAddDescription
};

export default connect(mapStateToProps, mapDispatchToProps)(HttpForm);
