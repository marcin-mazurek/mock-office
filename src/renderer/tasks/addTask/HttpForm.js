import React from 'react';
import { connect } from 'react-redux';
import { init as initAddTask } from './actions';
import { getSelectedServerDetails } from '../../servers/selectors';

export class HttpForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleBlockingChange = this.handleBlockingChange.bind(this);
    this.handleRequirementsChange = this.handleRequirementsChange.bind(this);
    this.handlePayloadChange = this.handlePayloadChange.bind(this);
    this.handleDelayChange = this.handleDelayChange.bind(this);
    this.state = {};
  }

  handleBlockingChange(event) {
    this.setState({
      blocking: event.target.checked
    });
  }

  handleRequirementsChange(event) {
    this.setState({
      requirements: event.target.value
    });
  }

  handlePayloadChange(event) {
    this.setState({
      taskPayload: event.target.value
    });
  }

  handleDelayChange(event) {
    this.setState({
      delay: parseInt(event.target.value, 10)
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const delay = parseInt(this.state.delay, 10);
    let requirements;

    try {
      if (this.state.requirements) {
        requirements = JSON.parse(this.state.requirements);
      }
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    let taskPayload;

    try {
      taskPayload = JSON.parse(this.state.taskPayload);
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    this.props.initAddTask(this.props.queueId, {
      blocking: this.state.blocking,
      delay,
      requirements,
      taskPayload
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <label htmlFor="blocking">Blocking:</label>
          <input
            type="checkbox"
            name="blocking"
            onChange={this.handleBlockingChange}
            value={this.state.blocking}
          />
        </div>
        <div>
          <label htmlFor="delay">Delay:</label>
          <input type="number" name="delay" onChange={this.handleDelayChange} />
        </div>
        <div>
          <label htmlFor="requirements">Requirements:</label>
          <div>
            <textarea
              name="requirements"
              cols="30"
              rows="10"
              onChange={this.handleRequirementsChange}
            />
          </div>
        </div>
        <div>
          <label htmlFor="payload">Payload:</label>
          <div>
            <textarea name="payload" cols="30" rows="10" onChange={this.handlePayloadChange} />
          </div>
        </div>
        <div>
          <button type="submit">
            Submit
          </button>
        </div>
      </form>
    );
  }
}

HttpForm.propTypes = {
  initAddTask: React.PropTypes.func.isRequired,
  queueId: React.PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  queueId: getSelectedServerDetails(state).queue
});

const mapDispatchToProps = {
  initAddTask
};

export default connect(mapStateToProps, mapDispatchToProps)(HttpForm);
