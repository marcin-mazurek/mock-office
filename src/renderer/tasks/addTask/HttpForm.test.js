import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HttpForm } from './HttpForm';

test('HttpForm should render', () => {
  const props = {
    initAddTask: () => {},
    queueId: 'some id'
  };
  const wrapper = shallow(<HttpForm {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
