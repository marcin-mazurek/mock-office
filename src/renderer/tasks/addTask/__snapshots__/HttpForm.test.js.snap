exports[`HttpForm HttpForm should render 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm de 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm defau 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default  1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default s 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default sh 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default sha 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default sna 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default snap 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`HttpForm default snapshot 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`test HttpForm should render 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;
